integration_test/sql/migration.exs

```diff
@@ -71,6 +71,43 @@ defmodule Ecto.Integration.MigrationTest do
     end
   end
 
+  defmodule AlterColumnFromMigration do
+    use Ecto.Migration
+
+    def change do
+      create table(:modify_from_authors, primary_key: false) do
+        add :id, :integer, primary_key: true
+      end
+      create table(:modify_from_posts) do
+        add :author, references(:modify_from_authors, type: :integer)
+      end
+
+      if direction() == :up do
+        execute "INSERT INTO modify_from_authors (id) VALUES (1)"
+        execute "INSERT INTO modify_from_posts (author) VALUES (1)"
+      end
+
+      alter table(:modify_from_posts) do
+        # remove the constraints modify_from_posts_author_fkey
+        modify :author, :integer, from: references(:modify_from_authors, type: :integer)
+      end
+      alter table(:modify_from_authors) do
+        modify :id, :bigint, from: :integer
+      end
+      case PoolRepo.__adapter__ do
```

Comment:
We avoid doing this in tests. Since you have added a tag to the test as `@tag :modify_column_with_from`, we can go to `integration_test/mysql/test_helper.exs` and exclude that tag. But please make sure to leave a TODO on top of the tag saying we can enable the test once we require 5.7+.